fastestimator/trace/io/csv_logger.py

```diff
@@ -12,66 +12,45 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 # ==============================================================================
-from typing import Optional, List
-
 import os
+from typing import List, Set, Union
+
+import pandas as pd
 
 from fastestimator.trace import Trace
-from fastestimator.util.util import is_number, to_list
 from fastestimator.util import Data
+from fastestimator.util.util import to_list
 
 
 class CSVLogger(Trace):
-    """Log monitored quantity in CSV file manner
+    """Log monitored quantity in CSV file
     Args:
         filename: Output filename.
-        monitor_names: List of key names to monitor. The names can be
-            {"mode", "epoch", "train_step", or output names that other traces
-            create}. If None, it will record all. Defaults to None.
-        separator: Seperator for numbers. Defaults to ", ".
-        append: If true, it will write csv file in append mode. Otherwise, it
-            will overwrite the existed file. Defaults to False.
-        mode: Restrict the trace to run only on given modes {'train', 'eval',
-            'test'}. None will always execute. Defaults to 'eval'.
+        monitor_names: List of key names to monitor.
+        mode: Restrict the trace to run only on given modes. None will always execute.
     """
     def __init__(self,
                  filename: str,
-                 monitor_names: Optional[List[str]] = None,
-                 separator: str = ", ",
-                 append: bool = False,
-                 mode: str = "eval"):
-        self.keys = monitor_names if monitor_names is None else to_list(monitor_names)
-        super().__init__(inputs="*" if self.keys is None else monitor_names, mode=mode)
-        self.separator = separator
+                 monitor_names: Union[List[str], str],
```

Comment:
making monitor names mandatory seems less desirable than previous version

Reply:
well, anyone who uses the trace must have a reason right?  then the reason is mandatory.

Reply:
what if the reason is I want to record everything to csv?

Reply:
done